src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs

```diff
@@ -41,6 +41,11 @@ public class RedisCacheOptions : IOptions<RedisCacheOptions>
     /// </summary>
     public Func<ProfilingSession>? ProfilingSession { get; set; }
 
+    /// <summary>
+    /// Support the "force reconnect" pattern as described <a href="https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-best-practices-connection#using-forcereconnect-with-stackexchangeredis">here</a>, which can help mitigate against unstable redis connections.
+    /// </summary>
+    public bool UseForceReconnect { get; set; }
```

Comment:
I agree with @BrennanConroy that if this is temporary, it might make more sense to use something like an AppContext switch rather than add public API. If we feel public API is warranted, we should take it through [api review](https://github.com/dotnet/aspnetcore/blob/main/docs/APIReviewProcess.md).

This logic is fairly complicated. I get that making this fully generic and decorating something like IConnectionMultiplexer might be too much, but could we create an internal shared-source API that participates in calling OnRedisError at the right times that can be used by our other redis-backed libraries?

Reply:
App-context switch would be fine; I can tweak that if it avoids the problem

Reply:
app-context switch implemented